Weaver/Weaver/Weaver.cs

```diff
@@ -38,7 +38,24 @@ class WeaverLists
 
     class Weaver
     {
-        // UNetwork types
+        public static WeaverLists WeaveList { get; private set; }
+        public static AssemblyDefinition CurrentAssembly { get; private set; }
+        public static ModuleDefinition CorLibModule { get; private set; }
+        public static AssemblyDefinition UnityAssembly { get; private set; }
+        public static AssemblyDefinition NetAssembly { get; private set; }
+        public static bool WeavingFailed { get; set; }
+        public static bool GenerateLogErrors { get; set; }
+
+        // private properties
+        static bool m_debugLogEnabled = true;
+
+        // this is used to prevent stack overflows when generating serialization code when there are self-referencing types.
+        // All the utility classes use GetWriteFunc() to generate serialization code, so the recursion check is implemented there instead of in each utility class.
+        // A NetworkBehaviour with the max SyncVars (64) can legitimately increment this value to 65 - so max must be higher than that
+        const int m_kMaxRecursionCount = 128;
```

Comment:
could you move all that stuff below again so it doesn't all show up as changes in the PR?

Reply:
Only private properties or also public ones?

Reply:
nevermind the moving. but please name the vars here
DebugLogEnabled
and
MaxRecursionCount
and
RecursionCount

the m_ prefix is kinda weird for static variables, usually people use s_ for that - or no prefix at all.

Reply:
https://github.com/vis2k/Mirror/pull/455/commits/daf73578e3d73af87cb26f31be71055bd6ae9a7f